src/bootstrap/src/core/build_steps/gcc.rs

```diff
@@ -112,16 +114,54 @@ impl Step for Gcc {
             return true;
         }
 
-        command(root.join("contrib/download_prerequisites")).current_dir(&root).run(builder);
-        command(root.join("configure"))
+        // GCC creates files (e.g. symlinks to the downloaded dependencies)
+        // in the source directory, which does not work with our CI setup, where we mount
+        // source directories as read-only on Linux.
+        // Therefore, as a part of the build in CI, we first copy the whole source directory
+        // to the build directory, and perform the build from there.
+        let src_dir = if CiEnv::is_ci() {
+            let src_dir = builder.gcc_out(target).join("src");
+            if src_dir.exists() {
+                builder.remove_dir(&src_dir);
+            }
+            builder.create_dir(&src_dir);
+            builder.cp_link_r(&root, &src_dir);
+            src_dir
+        } else {
+            root
+        };
+
+        command(src_dir.join("contrib/download_prerequisites")).current_dir(&src_dir).run(builder);
+        let mut configure_cmd = command(src_dir.join("configure"));
+        configure_cmd
             .current_dir(&out_dir)
+            // On CI, we compile GCC with Clang.
+            // The -Wno-everything flag is needed to make GCC compile with Clang 19.
+            // Note: this disables debug information for GCC.
+            // We might want to enable it in the future in an opt-in manner.
```

Comment:
Assuming we are not uploading CI GCC anywhere, why not apply this only in CI to avoid affecting developers?

Reply:
> why not apply this only in CI to avoid affecting developers?

or, for better, apply only when compiling GCC with `clang`

Reply:
I don't want to do clang detection here, that might not be so trivial. Eventually, once we add `[gcc]` section to `config.toml`, we will want to generate these flags ourselves anyway. Right now, we don't have that configuration yet, so I can use the most general option, i.e. `-Wno-everything -O2 -g`, which just adds `-Wno-everything` to the previous default (`-O2 -g`). Would that be fine for you?